plugins/russian/authortoday.ts

```diff
@@ -1,538 +1,529 @@
+  site = "https://author.today";
+  version = "1.0.0";
+
+  async popularNovels(
+    pageNo: number,
+    { showLatestNovels, filters }: Plugin.PopularNovelsOptions,
+  ): Promise<Plugin.NovelItem[]> {
+    let url = apiUrl + "v1/catalog/search?page=" + pageNo;
+    if (filters?.genre?.value) {
+      url += "&genre=" + filters.genre.value;
+    }
+
+    url +=
+      "&sorting=" +
+      (showLatestNovels ? "recent" : filters?.sort?.value || "popular");
+
+    url += "&form=" + (filters?.form?.value || "any");
+    url += "&state=" + (filters?.state?.value || "any");
+    url += "&series=" + (filters?.series?.value || "any");
+    url += "&access=" + (filters?.access?.value || "any");
+    url += "&promo=" + (filters?.promo?.value || "hide");
+
+    const result = await fetchApi(url, {
+      headers: {
+        Authorization: token,
+      },
+    });
+    const json = (await result.json()) as response;
+    const novels: Plugin.NovelItem[] = [];
+
+    if (json.code === "NotFound") {
+      return novels;
+    }
+
+    json?.searchResults?.forEach((novel) =>
+      novels.push({
+        name: novel.title,
+        cover: novel.coverUrl
+          ? "https://cm.author.today/content/" + novel.coverUrl
+          : defaultCover,
+        url: this.site + "/work/" + novel.id,
+      }),
+    );
+
+    return novels;
+  }
+  async parseNovelAndChapters(novelUrl: string): Promise<Plugin.SourceNovel> {
+    const workID = novelUrl.split("/")[4];
+    const result = await fetchApi(`${apiUrl}v1/work/${workID}/details`, {
+      headers: {
+        Authorization: token,
+      },
+    });
+
+    const book = (await result.json()) as responseBook;
+    const novel: Plugin.SourceNovel = {
+      url: novelUrl,
+      name: book.title,
+      cover: book.coverUrl ? book.coverUrl.split("?")[0] : defaultCover,
+      genres: book.tags?.join(", "),
+      summary: "",
+      author:
+        book.originalAuthor ||
+        book.authorFIO ||
+        book.coAuthorFIO ||
+        book.secondCoAuthorFIO ||
+        book.translator ||
+        "",
+      status: book.isFinished ? NovelStatus.Completed : NovelStatus.Ongoing,
+    };
+
+    if (book.annotation) novel.summary += book.annotation + "\n";
+    if (book.authorNotes) novel.summary += "Примечания автора:\n" + book.authorNotes;
+
+    // all chapters
+    const chaptersRaw = await fetchApi(`${apiUrl}v1/work/${workID}/content`, {
+      headers: {
+        Authorization: token,
+      },
+    });
+
+    const chaptersJSON = (await chaptersRaw.json()) as ChaptersEntity[];
+    const chapters: Plugin.ChapterItem[] = [];
+    chaptersJSON.forEach((chapter, chapterIndex) => {
+      if (chapter.isAvailable && !chapter.isDraft) {
+        chapters.push({
+          name: chapter.title || 'Глава ' + (chapterIndex + 1),
+          url: `${apiUrl}v1/work/${workID}/chapter/${chapter.id}/text`,
+          releaseTime: dayjs(chapter.publishTime || chapter.lastModificationTime).format("LLL"),
```

Comment:
could you check if this work in app? hermes engine is stupid with some date format.

Reply:
It's working. 
![Screenshot_20240204-075331](https://github.com/LNReader/lnreader-sources/assets/58046032/1aaba9ec-1c09-4382-b330-54d2ac14d8d7)
![Screenshot_20240204-075350_1](https://github.com/LNReader/lnreader-sources/assets/58046032/377d1dd8-231f-4b86-8c1b-97b7da21ae2b)